fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/PlanSerializationContext.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * PlanSerializationContext.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record;
+
+import com.apple.foundationdb.record.PlanHashable.PlanHashMode;
+import com.apple.foundationdb.record.RecordQueryPlanProto.PPlanReference;
+import com.apple.foundationdb.record.query.plan.cascades.IdentityBiMap;
+import com.apple.foundationdb.record.query.plan.plans.RecordQueryPlan;
+import com.apple.foundationdb.record.query.plan.serialization.DefaultPlanSerializationRegistry;
+import com.apple.foundationdb.record.query.plan.serialization.PlanSerializationRegistry;
+import com.google.common.base.Verify;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+/**
+ * TBD.
```

Comment:
At some point, we probably do want to expand on this. We should probably mark this class as `INTERNAL`, I think